test/nn/test_module_hooks.py

```diff
@@ -587,22 +588,45 @@ def hook_with_module(
             nonlocal hook_called
             hook_called += 1
 
+        def hook_with_module_returns_state_dict(
+            module,
+            state_dict,
+            prefix,
+            local_metadata,
+            strict,
+            missing_keys,
+            unexpected_keys,
+            error_msgs,
+        ):
+            self.assertEqual(m_state_dict, state_dict)
+            self.assertTrue(m_load is module)
+            nonlocal hook_called
+            hook_called += 1
+            return m_for_state_dict_hook
+
         hook_called = 0
+        # Test private API since this sets with_module=False which diverges from public API
         m_load._register_load_state_dict_pre_hook(hook_without_module)
         m_load.load_state_dict(m_state_dict)
         self.assertEqual(1, hook_called)
 
         hook_called = 0
-        m_load._register_load_state_dict_pre_hook(hook_with_module, True)
+        m_load.register_load_state_dict_pre_hook(hook_with_module)
```

Comment:
Do we want to keep one test that ensures that the with_module=True option in the private API continue to work well?

Reply:
Hm but the public API calls into the private API with True 😅 , but anyway sure